app/views/projects/tree/show.html.haml

```diff
@@ -3,12 +3,12 @@
 = content_for :meta_tags do
   - if current_user
     = auto_discovery_link_tag(:atom, namespace_project_commits_url(@project.namespace, @project, @ref, format: :atom, private_token: current_user.private_token), title: "#{@project.name}:#{@ref} commits")
-
 = render 'projects/last_push'
 
 - if can? current_user, :download_code, @project
   .tree-download-holder
     = render 'projects/repositories/download_archive', ref: @ref, btn_class: 'btn-group pull-right hidden-xs hidden-sm', split_button: true
+= render 'projects/find_file_link'
```

Comment:
We need a little bit of distance between the "Find File" and "Download zip" buttons. This works:

- Change `_find_file_link.html` to:

```haml
= link_to namespace_project_find_file_path(@project.namespace, @project, @ref), class: 'btn btn-grouped shortcuts-find-file', rel: 'nofollow' do
  = icon('search')
  %span Find File
```

- Change in `tree/show.html.haml` from:

```haml
- if can? current_user, :download_code, @project
  .tree-download-holder
    = render 'projects/repositories/download_archive', ref: @ref, btn_class: 'btn-group pull-right hidden-xs hidden-sm', split_button: true
= render 'projects/find_file_link'
```
to:
```haml
.pull-right
  = render 'projects/find_file_link'
  - if can? current_user, :download_code, @project
    = render 'projects/repositories/download_archive', ref: @ref, btn_class: 'hidden-xs hidden-sm btn-grouped', split_button: true
```